### PR TITLE
Fixes #595 Downsides of using Cluster Safe Option in Health Check

### LINKED ISSUE #595: Document considerations about health check REST endpoints
In https://docs.hazelcast.com/hazelcast/5.2/maintain-cluster/monitoring#health-check-and-monitoring we should explicitly document that health checks which include the "cluster safe" information can impact heavily loaded clusters. Copying from a [slack thread](https://hazelcast.slack.com/archives/C047ESQN80N/p1667405942257429?thread_ts=1667399913.834999&cid=C047ESQN80N):

> The usefulness of “cluster safety” as a health indicator is questionable in any cluster state other than PASSIVE because the cluster is in a dynamic state. Backups may have fallen temporarily behind (but that’s not unexpected especially for a write-heavy use case). On top of its limited usefulness, check for cluster safety triggers a lot of operations (growing linearly with the number of partitions and replicas).

### PR DESCRIPTION
**Pull request by angelasimms**
Fixes #595.

### CODE REVIEW

File: docs/modules/maintain-cluster/pages/monitoring.adoc [hunk 1]
```diff
@@ -1061,6 +1065,8 @@ The following parameters can be used:
 |Health check operation. It can be `all`, `node-state`,
 `cluster-state`, `cluster-safe`, `migration-queue-size` and `cluster-size`.
 
+NOTE: The `clusterSafe` indicator is useful when your cluster is in a passive state. If the cluster is in an active state, the indicator value continually changes due to the dynamic state of the cluster. Also, checking the cluster safety triggers additional operations for each partition and replica, which means operations increase with increased load and may impact cluster performance.
```
Comment (vbekiaris) [inline docs/modules/maintain-cluster/pages/monitoring.adoc:1068]: Maybe "Also, checking the cluster safety triggers additional operations for each partition and replica. Frequent cluster safety checks in a cluster under load may impact cluster performance." ? Basically don't query cluster safety every 5 seconds :)

**Review by vbekiaris — APPROVED** [review]
thanks for the enhancement, added a minor comment

File: docs/modules/maintain-cluster/pages/monitoring.adoc (hunk 1 above)
Comment (angelasimms) [reply docs/modules/maintain-cluster/pages/monitoring.adoc]: Updated as requested.

**Review by vbekiaris — APPROVED** [review]

**Comment by vbekiaris** [comment]
thanks for the improvements @angelasimms

**Pull request merged**